boot/boot.go

```diff
@@ -463,12 +480,26 @@ func updateManagedBootConfigForBootloader(dev snap.Device, mode, gadgetSnapOrDir
 		}
 		return false, err
 	}
+
 	// boot config update can lead to a change of kernel command line
-	_, err = observeCommandLineUpdate(dev.Model(), commandLineUpdateReasonSnapd, gadgetSnapOrDir, cmdlineAppend)
+	cmdlineChange, err := observeCommandLineUpdate(dev.Model(), commandLineUpdateReasonSnapd, gadgetSnapOrDir, cmdlineAppend)
 	if err != nil {
 		return false, err
 	}
-	return tbl.UpdateBootConfig()
+
+	if cmdlineChange {
+		candidate := true
+		if err := updateCmdlineVars(tbl, gadgetSnapOrDir, cmdlineAppend, candidate); err != nil {
```

Comment:
Why we need to do this now but not in the past? Was it a bug?

Reply:
Not really. Though not really sure of all the corner cases. If we update the boot assets, they might use a `grub.cfg` with an updated default kernel command line. After this change, since we do not uses `snapd_extra_cmdline_args` anymore, this default is not used. We take the default from `bootloader/assets/grub.go`. Depending on the version of the assets, the default is different. So to keep the same behavior no matter of the version of the assets installed, we have to update `snapd_full_cmdline_args` with the default arguments matching the version. Even if the extra command line has not changed.

Reply:
I think what I said might be confusing. This code happens only when snapd is updated. And it might bring a new `grub.cfg`.

Reply:
Ok, it is clear to me know, thanks for the explanation.